src/lambda_handler.py

```diff
@@ -34,54 +164,253 @@ def task_default(event):
+    description: str
+    filename: str
+    # I'd prefer to define ip_set as Set[Union[IPv4Address, IPv6Address]],
+    # but ipaddress.collapse_addresses() uses the type variable "_N" while
+    # ip_network returns Union[IPv4Network, IPv6network] which causes
+    # the mypy pre-commit hook to throw this error:
+    #   Value of type variable "_N" of "collapse_addresses" cannot be
+    #   "Union[IPv4Network, IPv6Network]"
+    # My solution to this problem is to simply define ip_set as Set[Any].
+    # For a similar issue and discussion, see
+    # https://github.com/python/typeshed/issues/2080
+    ip_set: Set[Any]
+    static_ips: List[str]
+
+
+def validate_event_data(
+    event: Dict[str, Any]
+) -> Tuple[Dict[str, Any], bool, List[str]]:
+    """Validate the event data and return a tuple containing the validated event, a boolean result (True if valid, False if invalid), and a list of error message strings."""
+    result = True
+    errors = []
+
+    # Account ID checks
+    if "account_ids" not in event:
+        errors.append('Missing required key "account_ids" in event.')
+    elif event["account_ids"] == []:
+        errors.append('"account_ids" must be a non-empty list.')
     else:
-        contents: str = event.get("contents", "Hello from AWS Lambda!")
-        logger.info(
-            'Creating output using "%s" with contents "%s"', character, contents
-        )
-        result["message"] = cowsay.get_output_string(character, contents)
+        account_ids: List[str] = event["account_ids"]
+        try:
+            # Ensure account_ids is a list of strings
+            if type(account_ids) is not list:
+                account_ids = [str(account_ids)]
+            else:
+                account_ids = [str(e) for e in account_ids]
 
-    return result
+            # Verify that each provided AWS account ID is 12 digits
+            invalid_account_ids = []
+            for account_id in account_ids.copy():
+                if not re.match(r"^\d{12}$", account_id):
+                    account_ids.remove(account_id)
+                    invalid_account_ids.append(account_id)
 
+            if invalid_account_ids:
+                errors.append(
+                    f'Invalid account ID(s) provided: "{", ".join(invalid_account_ids)}" - ID must be 12 digits.'
+                )
+        except Exception:
+            errors.append("account_ids must be a list of strings.")
+        event["account_ids"] = account_ids
 
-def task_divide(event) -> dict[str, Union[Optional[float], bool]]:
-    """Divide one number by another and provide the result."""
-    result: dict[str, Union[Optional[float], bool]] = {"result": None, "success": True}
-    numerator: str = event.get("numerator", None)
-    denominator: str = event.get("denominator", None)
+    # Bucket name checks
+    if "bucket_name" not in event:
+        errors.append('Missing required key "bucket_name" in event.')
+    elif type(event["bucket_name"]) != str:
```

Comment:
Same comment about `isinstance()` vs `type()`.

```suggestion
    elif not isinstance(event["bucket_name"], str):
```